architecture-seat-adjuster.md

```diff
@@ -0,0 +1,82 @@
+The seat adjuster application interacts with the vehicle through a *Vehicle Abstraction Layer* created by the KUKSA Databroker,
+which uses the [Vehicle Signal Specification (VSS)](https://covesa.github.io/vehicle_signal_specification/)
+to express the current value and in case of actuators also the desired state of the vehicle signal.
+By developing against the abstraction layer, the application becomes independent from the actual physical seat.
+
+To control the position of the driver seat, the seat adjuster sets the target value of the `Vehicle.Cabin.Seat.Row1.Pos1.Position` signal in the KUKSA Databroker.
+
+In addition, we have the seat service that subscribes to this signal and is responsible for realizing this target state in the physical world
+by interacting with the responsible ECUs, e.g., through CAN.
+Once the seat has moved, the seat service updates the current value of the `Vehicle.Cabin.Seat.Row1.Pos1.Position` signal.
+For our initial setup, the seat service emulates the hardware interaction and updates the current value based on the target value.
+In the diagrams, we still include the communication over CAN to show the whole interaction from an actual vehicle.
+
+```mermaid
+flowchart TB
+    client[Client]
+    anotherClient[Another Client]
+    seatadjuster[Seat Adjuster]
+    databroker[(KUKSA<br>Databroker)]
+    mqttRequest[[MQTT topic<br>seatadjuster/setPosition/request]]
+    %% mqttResponse[[MQTT topic<br>seatadjuster/setPosition/response]]
+    %% mqttCurrent[[MQTT topic<br>seatadjuster/currentPosition]]
+    seatservice[Seat Service]
+    canbus[CAN-Bus and Seat ECU]
+
+    client -- "JSON Request: position, requestId" --> mqttRequest
+    anotherClient -.-> mqttRequest
+    mqttRequest --> seatadjuster
+    seatadjuster -- "Set Target<br>Vehicle.Cabin.Seat.Row1.Pos1.Position<br>gRPC" --> databroker
+    databroker <-- "Set Current<br>Vehicle.Cabin.Seat.Row1.Pos1.Position<br><br>Notify subscriber of changed target<br>Vehicle.Cabin.Seat.Row1.Pos1.Position" --> seatservice
+    seatservice -- "CAN Frame: SECU1_CMD1<br>CAN-ID 0x705" --> canbus
+```
+
+As interface to the user, we assume a **client** that can, for example, be a local app in the infotainment domain with a user interface
+or an off-board application sending the request from a backend.
+Either way, the client controls our seat adjuster application through a JSON encoded message over MQTT using the topic `setPosition/request`.
+
+An example request looks like this:
+
+```bash
+mosquitto_pub -t seatadjuster/setPosition/request
```

Comment:
this assumes having `mosquitto_pub` available which is for example not on our velocitas devcontainer

Reply:
True but in Leda it is preinstalled. Should I add a note saying that it needs to be installed?